cli-plugins/apps/how-to-guides.hbs.md

```diff
@@ -0,0 +1,611 @@
+It allows to list folders (these must not end with the system separator, e.g. `/` or `\`), files and have comments which start with `#`. If the file contains files or directories that are not in the source code, they are ignored.
+
+**Example of a .tanzuignore file**
+```bash
+    .tanzuignore # must contain itself in order to be ignored
+    # This is a comment
+    this/is/a/folder/to/exclude
+
+    this-is-a-file.ext
+```
+
+<!-- ## <a id='maximize-efficiency'> Leveraging ENVs to maximize efficiency
+
+## <a id='param-yaml-usage'> --param-yaml
+
+## <a id='app-usage'> --app
+
+## <a id='type-usage'> --type
+
+## <a id='labels-anns-env'> --labels, --annotations, --env, --build-env -->
+
+## <a id='dry-run-usage'> --dry-run
+
+The main goal of `--dry-run` flag is to prepare all the steps to submit a workload to the cluster and stop before sending it, showing
+an output of the final structure of the workload.
+
+For example, when applying a workload from Git source:
+
+```bash
+tanzu apps workload apply rmq-app --git-repo https://github.com/jhvhs/rabbitmq-sample --git-branch main --service-ref "rmq=rabbitmq.com/v1beta1:RabbitmqCluster:example-rabbitmq-cluster-1" -t web --dry-run
+---
+apiVersion: carto.run/v1alpha1
+kind: Workload
+metadata:
+  creationTimestamp: null
+  labels:
+    apps.tanzu.vmware.com/workload-type: web
+  name: rmq-app
+  namespace: default
+spec:
+  serviceClaims:
+  - name: rmq
+    ref:
+      apiVersion: rabbitmq.com/v1beta1
+      kind: RabbitmqCluster
+      name: example-rabbitmq-cluster-1
+  source:
+    git:
+      ref:
+        branch: main
+      url: https://github.com/jhvhs/rabbitmq-sample
+status:
+  supplyChainRef: {}
+```
+
+This will allow the user to check how it's workload would be created/updated in the cluster.
+If there would be an error when trying to apply the workload, this would be shown with the `--dry-run` flag. 
+
+```bash
+tanzu apps workload create rmq-sample-app --git-repo https://github.com/jhvhs/rabbitmq-sample --git-branch main --service-ref "rmq=rabbitmq.com/v1beta1:RabbitmqCluster:example-rabbitmq-cluster-1" -t web --dry-run
+Error: workload "default/rmq-sample-app" already exists
+```
+
+## <a id='update-strategy-usage'> --update-strategy
+
+A workload specification can be changed using Apps CLI plug-in through the `--update-strategy` flag, which checks whether the current workload in the cluster needs to be merged or replaced, defaulting if not specified as `merge`.
+
+**Note**: the default behavior of `--update-strategy` will be deprecated in upcoming releases.
+
+This flag is used along with `--file` flag (which is required) and it has two behaviors:
+
+- `merge`, which does not delete fields in the workload. Most of these can be updated, but won't be taken out from the workload regardless they were deleted on the yaml specification.
+- `replace`, which changes the workload to be exactly as specified in the yaml definition.
+
+For example:
```

Comment:
Examples of the outcomes of both `merge` and `replace` update strategies are provided below: